src/monattack.cpp

```diff
@@ -4647,3 +4648,19 @@ bool mattack::dodge_check( monster *z, Creature *target ) {
     }
     return false;
 }
+
+void mattack::lay_egg( monster *z )
+{
+    monster &critter = *z;
+    const furn_t &furn = g->m.furn( critter.pos() ).obj();
+
+    if( !critter.is_dead() && critter.has_flag( MF_EGGLAYING ) && furn.has_flag( TFLAG_NEST ) &&
```

Comment:
> I don't have any idea what you mean by MF_SEEKS_NEST, and I can't find a way to make a lay_egg attack.

Your `MF_EGGLAYING` flag controls both the nest seeking and the egg laying. But that is not the proper way to do it, as one part of this behavior is just seeking nests and the other is laying eggs.
Seeking nests should happen in planning phase because it determines what the monster wants to do next.
Laying eggs should be an attack because it is what the monster actually does.

> I barely understand how the whole attack thing works and there isn't much documentation about it.

You have attacks declared in `monattack.h`, implemented in `monattack.cpp`, "named" in `monstergenerator.cpp` (in `init_attack` - just create a new entry), then bound to monster's special attack list in monster's json.
For this to be possible, all monster attacks need the same signature: `bool attack_name_here( monster *z );`.

You did most of that, but didn't do the `monstergenerator.cpp` part yet. This is why you're getting the error: the attack exists, but the game can't "find" it.